### ISSUE TITLE
Issue with `svelte-preprocess`

### ISSUE THREAD

**Issue by Rajaniraiyn** [parcel-bundler/lightningcss#418]
I am trying to add lightning css directly into `svelte-process` as one of embedded default transformers. It works but not completely as I couldn't able to find the documentation for `lightningcss` transform function other than few examples and TypeScript type definitions. I am seeking help as a support or proper documentation for `lightningcss` 

`svelte-preprocess` fork with `lightningcss` support: https://github.com/Rajaniraiyn/svelte-preprocess
Related Issue: https://github.com/parcel-bundler/lightningcss/issues/389

**Comment by devongovett** [comment]
There is quite a bit of documentation here and on other pages on the website: https://lightningcss.dev/docs.html What do you find missing?

**Comment by Rajaniraiyn** [comment]
yes there is a bit in website and remaining in typescript definitions but those type definitions are not clear for few types.
e.g.
```ts
export interface TransformOptions {
  ...
  /** An input source map to extend. */
  inputSourceMap?: string,
  ...
  /** Whether to compile this file as a CSS module. */
  cssModules?: boolean | CSSModulesConfig,
  /**
   * Whether to analyze dependencies (e.g. `@import` and `url()`).
   * When enabled, `@import` rules are removed, and `url()` dependencies
   * are replaced with hashed placeholders that can be replaced with the final
   * urls later (after bundling). Dependencies are returned as part of the result.
   */
  analyzeDependencies?: boolean | DependencyOptions,
  ...
}
```
in the above, all have to trail and error for,
- `inputSourceMap` that wether filename of source-map or string content of it.
- wether `cssModules` will also analyzes the imports (`@import` and `url()`)

If they are documented clearly, it would make developer experience much better
